backend/src/main/java/hu/flowacademy/worksheet/service/UserService.java

```diff
@@ -30,6 +34,16 @@ public User saveUser(@NonNull User user) throws ValidationException {
         return userRepository.save(user);
     }
 
+    public User update(Long id, User user) throws ValidationException {
+        validateUpdate(user);
+        User updatedUser = userRepository.findById(id).orElseThrow(() -> new ValidationException("The id is null: " + user.getId()));
```

Comment:
Javítsd az hibaüzenetet, mert nem csak akkor lehet hiba, ha az id null, hanem akkor is, ha nem valós. Használd itt a függvény paraméterben kapott ID-t, ne a user adattagját.